api/src/lib/interface.ts

```diff
@@ -138,6 +138,15 @@ export interface UpdatePublicationRequestBody {
     dataPermissionsStatementProvidedBy?: string;
     dataAccessStatement?: string;
     selfDeclaration?: boolean;
+    topics?: string[];
+}
+
+export interface UpdatePublicationData extends Omit<UpdatePublicationRequestBody, 'topics'> {
```

Comment:
I think it's a bit odd to first define the "topics" field on UpdatePublicationRequestBody and then extending that interface but omitting "topics" field which you just defined, just to change it's type to something that prisma expects. I would rather use UpdatePublicationRequestBody interface as it is, with the "topics" as a string[] and do that set mapping into the update service itself.

Reply:
That makes sense - I've removed this new interface and moved the topics manipulation to the service file.